wallet-new/src/Cardano/Wallet/API/WIP/Handlers.hs

```diff
@@ -39,14 +46,39 @@ deleteExternalWallet :: PassiveWalletLayer IO
 deleteExternalWallet _pwl _encodedRootPK =
     error "[CHW-106], Cardano Hardware Wallet feature, , delete external wallet, unimplemented yet."
 
+-- Matches the input InputGroupingPolicy with the Kernel's 'InputGrouping'
+toInputGrouping :: Maybe (V1 InputSelectionPolicy) -> InputGrouping
```

Comment:
This looks like a useful standalone function on its own. I suggest you simplify the type to be:

```
toInputGrouping :: V1 InputSelection -> InputGrouping
```

And you add this one to `module Cardano.Wallet.WalletLayer.Kernel.Conv`. Later, in this code, you would call it like so:

```
let inputGrouping = Kernel.Conv.toInputGrouping $ fromMaybe (V1 defaultInputSelectionPolicy) v1GroupingPolicy
```